src/Symfony/Bundle/FrameworkBundle/Console/Descriptor/JsonDescriptor.php

```diff
@@ -218,10 +218,10 @@ private function getContainerDefinitionData(Definition $definition, bool $omitTa
             if (\is_array($factory)) {
                 if ($factory[0] instanceof Reference) {
                     $data['factory_service'] = (string) $factory[0];
-                } elseif ($factory[0] instanceof Definition) {
-                    throw new \InvalidArgumentException('Factory is not describable.');
-                } else {
+                } elseif (!$factory[0] instanceof Definition) {
                     $data['factory_class'] = $factory[0];
+                } elseif ($factoryClass = $factory[0]->getClass()) {
+                    $data['factory_class'] = $factoryClass;
```

Comment:
```php
if ($factory[0] instanceof Reference) {
    $data['factory_service'] = (string) $factory[0];
} elseif ($factory[0] instanceof Definition) {
    $data['factory_class']->getClass();
} else {
    $data['factory_class'] = $factory[0];
}
```

Maybe like this instead?

Reply:
I originally did it like you suggest but changed because a definition class could be null. Don't know how we should handle it or if it can happen at all 🤔

Reply:
But changing the order we check each condition in the code wouldn't change the produced result for these cases, would it?

Reply:
Yep but there’s one more change here: checking if the definition class is null or not. I was under the impression we don’t want the `factory_class` key if its value is `null`, that’s why I wrote the condition like this.

Reply:
Hm, you are right indeed. But maybe we should display something different when that the case?

Maybe like this:

```php
if ($factory[0] instanceof Reference) {
    $data['factory_service'] = (string) $factory[0];
} elseif ($factory[0] instanceof Definition) {
    $data['factory_class'] = $definition->getClass() ?: 'not configured';
} else {
    $data['factory_class'] = $factory[0];
}
```

Reply:
Doesn’t matter to me; I don’t even know if it can happen :sweat_smile: 

I’ll update the PR!